### PR TITLE
Support for FunAsSeq + HOWTO on type annotations

### PR DESCRIPTION
**Pull request by konnov**
This PR closes the issue #223. The type checker now understands the type conversion operator `FunAsSeq`. That was quite easy to add, so I diverged in writing the HOWTO on type annotations.

- [x] Tests added for any new code
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [x] Documentation added for any new functionality
- [x] Entry added to [UNRELEASED.md](./UNRELEASED.md) for any new functionality

### CODE REVIEW

File: docs/src/HOWTOs/howto-write-type-annotations.md [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+This HOWTO gives you concrete steps to extend TLA+ specifications with type
+annotations. You can find the detailed syntax of type annotations in
+[ADR002][]. The first rule of writing type annotations:
+
+*Do not to write any annotations at all, until the type checker [Snowcat][] is
+asking you to write a type annotation.*
+
+Snowcat is quite smart, and it will most probably find most of the types for
+you.
```
Comment (shonfeder) [inline docs/src/HOWTOs/howto-write-type-annotations.md:16]: Given the very first example, where It cannot deduce that `hr` is a nat even tho it is clearly a nat from the very next line, I think this line may give unrealistic expectations.

File: docs/src/HOWTOs/howto-write-type-annotations.md [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+
+## Recipe 5: functions as sequences
+
+Check the example [Queens.tla][] from the repository of TLA+ examples.  It has
+85 lines of code, so we do not include it here. Similar to the previous
+sections, we annotate constants and variables:
+
+```tla
+CONSTANT \* @type: Int;
+         N
```
Comment (shonfeder) [inline docs/src/HOWTOs/howto-write-type-annotations.md:261]: To be consistent with the formatting on line 116: 

```suggestion
CONSTANT 
         \* @type: Int;
         N
```

File: docs/src/SUMMARY.md [hunk 3]
```diff
@@ -22,13 +22,17 @@
     - [Five minutes of theory](./apalache/theory.md)
 - [TLC Configuration Files](./apalache/tlc-config.md)
 - [Types and Annotations](./apalache/types-and-annotations.md)
-- [Snowcat: the New Type Checker](./apalache/typechecker-snowcat.md)
+- [The New Type Checker Snowcat](./apalache/typechecker-snowcat.md)
```
Comment (shonfeder) [inline docs/src/SUMMARY.md:25]: I don't think this phrasing will age well. How about:

```suggestion
- [The Snowcat Type Checker](./apalache/typechecker-snowcat.md)
```

File: docs/src/SUMMARY.md [hunk 4]
```diff
@@ -22,13 +22,17 @@
...
 - [Supported Features](./apalache/features.md)
 - [TLA+ Preprocessing](./apalache/preprocessing.md)
 - [Fine Tuning](./apalache/tuning.md)
 - [KerA: kernel logic of actions](./apalache/kera.md)
 - [Assignments in Apalache](./apalache/assignments.md)
 
+# HOWTOs
+
+- [How to write type annotations](./HOWTOs/howto-write-type-annotations.md)
```
Comment (shonfeder) [inline docs/src/SUMMARY.md:34]: I wonder if this shouldn't just be part of the snowcat sections? Perhaps snowcat needs it's own directory and sub-chapters?

Lots of things int he manual describe how to do things: how to install, how to fine-tune, how to configure. I feel like howto sections often become weird grabbags, and it'd bet cleaner just to structure the manual by topic.

File: docs/src/HOWTOs/howto-write-type-annotations.md [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+asking you to write a type annotation.*
+
+Snowcat is quite smart, and it will most probably find most of the types for
+you.
+
+## Recipe 1: Recipe variables
+
+Consider the example [HourClock.tla][] from [Specifying Systems][]:
+
+```tla
```
Comment (shonfeder) [inline docs/src/HOWTOs/howto-write-type-annotations.md:22]: Should you maybe use the include syntax that we use elsewhere, to ensure that the manual? It helps ensure that the docs will exactly match whatever is in the source file: 

https://github.com/informalsystems/apalache/blob/shon/integration-521-rebase-test/docs/src/apalache/example.md#L6-L7

**Review by shonfeder — APPROVED** [review]
LGTM (suggestion not withstanding :)).

File: docs/src/HOWTOs/howto-write-type-annotations.md (hunk 1 above)
Comment (konnov) [reply docs/src/HOWTOs/howto-write-type-annotations.md:16]: That's probably true. The problem is that declarations are analyzed in isolation. That is good and bad. But sure, I will rephrase it.

File: docs/src/SUMMARY.md (hunk 4 above)
Comment (konnov) [reply docs/src/SUMMARY.md:34]: Right. We have to restructure the docs. The reason why I called it a HOWTO is that I wanted to try this approach: https://documentation.divio.com

Comment (konnov) [reply docs/src/SUMMARY.md:34]: Looking at our current manual, it mixes a tutorial, HOWTOs and explanations. That's why it looks quite weird.

File: docs/src/HOWTOs/howto-write-type-annotations.md (hunk 5 above)
Comment (konnov) [reply docs/src/HOWTOs/howto-write-type-annotations.md:22]: Sure, that's nice. What is 55 here?

Comment (konnov) [reply docs/src/HOWTOs/howto-write-type-annotations.md:22]:
```
{{#include ../../../test/tla/y2k.tla::55}}
```

Comment (shonfeder) [reply docs/src/HOWTOs/howto-write-type-annotations.md:25]: 55 is the line number to start the inclusion from! You can also give a line number to end on. See the mdBook docs for more details :)

Comment (konnov) [reply docs/src/HOWTOs/howto-write-type-annotations.md:25]: Ah, that's nice!

**Pull request merged**